### PR TITLE
doc: examples for Near-related host functions

### PR DESCRIPTION
**Pull request by denbite**
Make it straightforward for developers how host functions can be imported and used

The final markdown doc page will appear as follows
image

### CODE REVIEW

**Comment by dj8yfo** [comment]
@denbite please take a look at https://github.com/nearuaguild/abstract-dao/pull/9

**Comment by dj8yfo** [comment]
@denbite please fix compilation errors. There's at least one, where `NearToken::from_yocto` is used instead of `NearToken::from_yoctonear`

**Comment by denbite** [comment]
@dj8yfo done, please review

**Comment by denbite** [comment]
@dj8yfo also just added a little text about the purpose of functions like `storage_write`, `storage_read`, etc, with a reference to the real-life example
must be helpful for devs to understand that those methods are typically used to upgrade/migrate state of a contract

**Comment by dj8yfo** [comment]
@race-of-sloths score 13

**Comment by race-of-sloths** [comment]
@denbite Thank you for your contribution! Your pull request is now a part of the [Race of Sloths](https://race-of-sloths.com)!
Weekly streak is on the road, smart strategy! Secure your streak with another PR!

[<picture>
    <source media="(prefers-color-scheme: dark)" srcset="https://race-of-sloths.com/images/bot-banner-dark.svg">
    <source media="(prefers-color-scheme: light)" srcset="https://race-of-sloths.com/images/bot-banner-light.svg">
    Shows inviting banner with latest news.
</picture>](https://race-of-sloths.com/latest-news)

[<picture>
    <source media="(prefers-color-scheme: dark)" srcset="https://badge.race-of-sloths.com/denbite?type=bot&pr=near/near-sdk-rs/1259&theme=dark">
    <source media="(prefers-color-scheme: light)" srcset="https://badge.race-of-sloths.com/denbite?type=bot&pr=near/near-sdk-rs/1259&theme=light">
    Shows profile picture for the author of the PR
</picture>](https://race-of-sloths.com/profile/denbite)


<details><summary>Current status: <i>waiting for scoring</i></summary>

We're waiting for maintainer to score this pull request with `@race-of-sloths score [0,1,2,3,5,8,13]` command. Alternatively, autoscoring [1,2] will be applied for this pull request

</details><details><summary>What is the Race of Sloths</summary>
<p>
Race of Sloths is a friendly competition where you can participate in challenges and compete with other open-source contributors within your normal workflow

For contributors:

- Tag @race-of-sloths inside your pull requests
- Wait for the maintainer to review and score your pull request
- Check out your position in the [Leaderboard](https://race-of-sloths.com/leaderboard)
- Keep weekly and monthly streaks to reach higher positions
- Boast your contributions with a dynamic picture of your [Profile](https://race-of-sloths.com/profile/denbite)

For maintainers:
- Score pull requests that participate in the Race of Sloths and receive a reward
- Engage contributors with fair scoring and fast responses so they keep their streaks
- Promote the Race to the point where the Race starts promoting you
- Grow the community of your contributors

Feel free to check [our website](https://race-of-sloths.com) for additional details!
</p>
</details> 

<details>
<summary>Bot commands</summary>

- For contributors
  - **Include a PR:** `@race-of-sloths include` to enter the Race with your PR
- For maintainers:
  - **Invite contributor** `@race-of-sloths invite` to invite the contributor to participate in a race or include it, if it's already a runner.
  - **Assign points:** `@race-of-sloths score [1/2/3/5/8/13]` to award points based on your assessment.
  - **Reject this PR:** `@race-of-sloths exclude` to send this PR back to the drawing board.
  - **Exclude repo:** `@race-of-sloths pause` to stop bot activity in this repo until `@race-of-sloths unpause` command is called

</details>

File: near-sdk/src/environment/env.rs
Comment (dj8yfo) [reply near-sdk/src/environment/env.rs:1674]: Module-level documentation at [near_sdk::store] should mention with links that following 
low-level calls are used in implemenations of abstractions the module defines:

`[near_sdk::env::storage_write]`
`[near_sdk::env::storage_read]`
`[near_sdk::env::storage_remove]`
`[near_sdk::env::storage_has_key]`

Comment (dj8yfo) [reply near-sdk/src/environment/env.rs:1674]: should mention usage in: 

https://github.com/near/near-sdk-rs/tree/master/near-contract-standards/src/upgrade/mod.rs

File: near-sdk/src/environment/env.rs
Comment (dj8yfo) [inline near-sdk/src/environment/env.rs:1703]: should mention usage in 
https://github.com/near/near-sdk-rs/tree/master/near-contract-standards/src/upgrade/mod.rs

File: near-sdk/src/environment/env.rs
Comment (dj8yfo) [inline near-sdk/src/environment/env.rs:1724]: should mention usage in 
https://github.com/near/near-sdk-rs/tree/master/near-contract-standards/src/upgrade/mod.rs

File: near-sdk/src/environment/env.rs
Comment (dj8yfo) [inline near-sdk/src/environment/env.rs:1727]: this call isn't used in the referenced repo, and also in a pending pr https://github.com/near-examples/update-migrate-rust/pull/2 
won't request change to remove this link, hoping that this and some of other suggestions will be adressed in some of subsequent prs

File: near-sdk/src/environment/env.rs
Comment (dj8yfo) [reply near-sdk/src/environment/env.rs:1674]: usage of this call is only present in pending pr https://github.com/near-examples/update-migrate-rust/pull/2.

File: near-sdk/src/environment/env.rs
Comment (dj8yfo) [reply near-sdk/src/environment/env.rs:1703]: usage of this call is only present in pending pr https://github.com/near-examples/update-migrate-rust/pull/2

File: near-sdk/src/environment/env.rs
Comment (dj8yfo) [inline near-sdk/src/environment/env.rs:1747]: this call isn't used in the referenced repo, and also in a pending pr https://github.com/near-examples/update-migrate-rust/pull/2 
won't request change to remove this link, hoping that this and some of other suggestions will be adressed in some of subsequent prs

File: near-sdk/src/environment/env.rs
Comment (dj8yfo) [inline near-sdk/src/environment/env.rs:1763]: this call isn't used in the referenced repo, and also in a pending pr https://github.com/near-examples/update-migrate-rust/pull/2 
won't request change to remove this link, hoping that this and some of other suggestions will be adressed in some of subsequent prs

File: near-sdk/src/environment/env.rs
Comment (dj8yfo) [inline near-sdk/src/environment/env.rs:1809]: should mention usage in 
https://github.com/near/near-sdk-rs/tree/master/near-contract-standards/src/fungible_token/storage_impl.rs
https://github.com/near/near-sdk-rs/tree/master/near-contract-standards/src/non_fungible_token/utils.rs
https://github.com/near/near-sdk-rs/tree/master/examples/fungible-token/tests/workspaces.rs

**Review by dj8yfo — APPROVED** [review]
@denbite it felt somewhat overwhelming to request changes for all found suggestions
with current pr not introducing anything particularly erroneous, 

so i'll extract those to an issue in case someone volunteers to further improve docs by adding links

File: near-sdk/src/environment/env.rs
Comment (PolyProgrammist) [reply near-sdk/src/environment/env.rs:1394]: how exactly do you want to mention original example?

**Pull request merged**
